src/components/LoadingOverlay/LoadingOverlay.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import {Loader} from '../Loader';
+import type {LoaderProps} from '../Loader';
+import {Box} from '../layout';
+import type {BoxProps} from '../layout';
+import {block} from '../utils/cn';
+
+import './LoadingOverlay.scss';
+
+const b = block('loading-overlay');
+
+export type OverlayBackground = 'base' | 'float';
+
+export interface LoadingOverlayProps extends BoxProps {
+    loaderProps?: LoaderProps;
+    background?: OverlayBackground;
+    loading?: boolean;
+    children?: React.ReactNode;
+    className?: string;
+    contentClassName?: string;
+}
+
+export function LoadingOverlay({
+    loaderProps,
+    background = 'base',
+    loading = false,
+    children,
+    className,
+    contentClassName,
+    ...boxProps
+}: LoadingOverlayProps) {
+    return (
+        <Box className={b(null, className)} {...boxProps}>
+            <div className={b('overlay', {visible: loading})}>
+                <div className={b('background', {type: background})} />
+                <Loader {...loaderProps} />
```

Comment:
how do you think, may be add ability to pass a component insted of hardcode Loader component?

```tsx
const LoadingOverlay = ({
  loaderSlot,
  size = 'm',
}: {loader: React.ReactNode}) => {
  return (
     ...
     {typeof loaderSlot === 'undefined' ? <Loader size={size} /> : loaderSlot}
     ...
  )
}
```
it is important to make it possible to pass `null` so that the display can be completely redefined.